src/core/struct_tree.js

```diff
@@ -259,16 +263,24 @@ class StructTreeRoot {
       parentTree.set("Nums", nums);
     }
 
-    const newNextkey = await StructTreeRoot.#writeKids({
+    const newNextKey = await StructTreeRoot.#writeKids({
       newAnnotationsByPage,
       structTreeRootRef,
+      structTreeRoot: this,
       kids: null,
       nums,
       xref,
       pdfManager,
+      newRefs,
       cache,
     });
-    structTreeRoot.set("ParentTreeNextKey", newNextkey);
+
+    if (newNextKey === -Infinity) {
```

Comment:
That feels like a slightly "surprising" fallback value to return from `#writeKids`, does other code actually depend on that exact value or could we perhaps change it to instead return `null` in that case?

Reply:
That said, I wonder if it's correct:
https://github.com/mozilla/pdf.js/blob/master/src/core/annotation.js#L658

wdyt ?

Reply:
Sorry, but I've got no idea what's correct or intended here unfortunately.

Is there any way to check what e.g. Adobe Reader generates for similar content, and thus follow that?

Reply:
I created a basic pdf and changed an from 0 to -1 and it doesn't work, when changing it for 79 works correctly.
So the numbers must be non-negative.